autocodesign/projectmanager/projecthelper.go

```diff
@@ -0,0 +1,448 @@
+package projectmanager
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/sliceutil"
+	"github.com/bitrise-io/go-xcode/autocodesign"
+	"github.com/bitrise-io/go-xcode/xcodeproject/schemeint"
+	"github.com/bitrise-io/go-xcode/xcodeproject/serialized"
+	"github.com/bitrise-io/go-xcode/xcodeproject/xcodeproj"
+	"github.com/bitrise-io/go-xcode/xcodeproject/xcscheme"
+	"howett.net/plist"
+)
+
+// ProjectHelper ...
+type ProjectHelper struct {
+	MainTarget       xcodeproj.Target
+	DependentTargets []xcodeproj.Target
+	UITestTargets    []xcodeproj.Target
+	XcProj           xcodeproj.XcodeProj
+	Configuration    string
+
+	buildSettingsCache map[string]map[string]serialized.Object // target/config/buildSettings(serialized.Object)
+}
+
+// NewProjectHelper checks the provided project or workspace and generate a ProjectHelper with the provided scheme and configuration
+// Previously in the ruby version the initialize method did the same
+// It returns a new ProjectHelper, whose Configuration field contains is the selected configuration (even when configurationName parameter is empty)
+func NewProjectHelper(projOrWSPath, schemeName, configurationName string) (*ProjectHelper, error) {
+	// Maybe we should do this checks during the input parsing
```

Comment:
I think it's better to do this once here instead of doing it in every step.

Reply:
Removed comment